src/libertem_holo/base/filters.py

```diff
@@ -345,3 +371,64 @@ def central_line_filter(
         return dest[slice_fft]
     else:
         return dest
+
+
+@numba.njit
+def butterworth_disk(shape, radius, order=12):
+    """
+    shape: output shape of the aperture
+
+    radius: radius in pixels
+
+    order: order of the butterworth filter
+    """
+    result = np.zeros(shape, dtype=np.float32)
+    cy = shape[0]/2
+    cx = shape[1]/2
+    for y in range(shape[0]):
+        for x in range(shape[1]):
+            d = np.sqrt((y-cy)**2 + (x-cx)**2)
+            result[y, x] = 1/np.sqrt(1 + np.pow((d/radius), 2*order))
+    return result
+
+
+@numba.njit
+def butterworth_line(shape, width, sb_position, length_ratio=0.9, order=12):
+    """
+    shape: output shape of the aperture
+
+    width: width of the line in pixels
+
+    order: order of the butterworth filter
+    """
+    result = np.zeros(shape, dtype=np.float32)
+    cy = shape[0] / 2 - 1
+    cx = shape[1] / 2 - 1
+    a = (sb_position[0] - cy) / (sb_position[1] - cx)
+    b = 1
+
+    # determine starting point:
+    sb_dist = np.sqrt((sb_position[0] - cy)**2 + (sb_position[1] - cx)**2)
+    length = sb_dist * (1 - length_ratio)
+
+    sb_sel = np.sign(sb_position[0] - cy)
+
+    # shift to starting point
+    cy += length * (sb_position[0] - cy) / sb_dist
+    cx += length * (sb_position[1] - cx) / sb_dist
+
+    c = -1/a
+    for y in range(shape[0]):
+        for x in range(shape[1]):
+            x0 = x - cx
+            y0 = y - cy
+            d = y0 - c * x0
+            xc = (d-c)/(a-b)
```

Comment:
There's still a division by zero in here; probably on this line (see CI)

Reply:
This was caused by `width=0`; in this case, this doesn't work to disable the line filter. I've added a `line_filter_width=None` case to the high-level function instead.